// openclaw/dist/extensions/whatsapp/api.js
import { r as resolveDefaultWhatsAppAccountId, t as listAccountIds } from "./account-ids-DG18xddX.js";
import { a as resolveWhatsAppAccount, i as listWhatsAppAuthDirs, n as hasAnyWhatsAppAuth, o as resolveWhatsAppAuthDir, r as listEnabledWhatsAppAccounts, s as resolveWhatsAppMediaMaxBytes, t as DEFAULT_WHATSAPP_MEDIA_MAX_MB } from "./accounts-BDqgpFEB.js";
import { a as normalizeWhatsAppMessagingTarget, i as normalizeWhatsAppAllowFromEntries, n as isWhatsAppUserTarget, o as normalizeWhatsAppTarget, r as looksLikeWhatsAppTargetId, t as isWhatsAppGroupJid } from "./normalize-target-BSx79XmS.js";
import { t as resolveWhatsAppOutboundTarget } from "./resolve-outbound-target-muaeoILm.js";
import { a as markdownToWhatsApp, i as jidToE164, n as assertWebChannel, o as resolveJidToE164, r as isSelfChatMode, s as toWhatsappJid, t as text_runtime_exports } from "./text-runtime-BvOFWrFo.js";
import { t as whatsappPlugin } from "./channel-rIHS8uRA.js";
import { n as WHATSAPP_LEGACY_OUTBOUND_SEND_DEP_KEYS } from "./outbound-base-DCa9y9QR.js";
import { t as whatsappCommandPolicy } from "./command-policy-O8sRnqUV.js";
import { n as resolveWhatsAppGroupToolPolicy, r as resolveWhatsAppGroupIntroHint, t as resolveWhatsAppGroupRequireMention } from "./group-policy-BDmO10Lm.js";
import { t as whatsappSetupPlugin } from "./channel.setup-DxyYDYKw.js";
import { t as DEFAULT_WEB_MEDIA_BYTES } from "./constants-w1BoViXF.js";
import { n as listWhatsAppDirectoryGroupsFromConfig, r as listWhatsAppDirectoryPeersFromConfig } from "./directory-config-CiANMx84.js";
import "./runtime-api-DAmPtBbk.js";
import { t as __testing } from "./access-control-DCqks_qR.js";
var CONFIG_DIR = text_runtime_exports.CONFIG_DIR;
var DEFAULT_LOG_DIR = text_runtime_exports.DEFAULT_LOG_DIR;
var DEFAULT_LOG_FILE = text_runtime_exports.DEFAULT_LOG_FILE;
var FILE_REF_EXTENSIONS_WITH_TLD = text_runtime_exports.FILE_REF_EXTENSIONS_WITH_TLD;
var SYSTEM_MARK = text_runtime_exports.SYSTEM_MARK;
var __test__ = text_runtime_exports.__test__;
var asNullableObjectRecord = text_runtime_exports.asNullableObjectRecord;
var asNullableRecord = text_runtime_exports.asNullableRecord;
var asOptionalObjectRecord = text_runtime_exports.asOptionalObjectRecord;
var asOptionalRecord = text_runtime_exports.asOptionalRecord;
var asRecord = text_runtime_exports.asRecord;
var bindAbortRelay = text_runtime_exports.bindAbortRelay;
var buildTimeoutAbortSignal = text_runtime_exports.buildTimeoutAbortSignal;
var chunkItems = text_runtime_exports.chunkItems;
var chunkMarkdownIR = text_runtime_exports.chunkMarkdownIR;
var clamp = text_runtime_exports.clamp;
var clampInt = text_runtime_exports.clampInt;
var clampNumber = text_runtime_exports.clampNumber;
var convertMarkdownTables = text_runtime_exports.convertMarkdownTables;
var createScopedExpiringIdCache = text_runtime_exports.createScopedExpiringIdCache;
var diagnosticLogger = text_runtime_exports.diagnosticLogger;
var displayPath = text_runtime_exports.displayPath;
var displayString = text_runtime_exports.displayString;
var ensureDir = text_runtime_exports.ensureDir;
var escapeRegExp = text_runtime_exports.escapeRegExp;
var fetchWithTimeout = text_runtime_exports.fetchWithTimeout;
var findCodeRegions = text_runtime_exports.findCodeRegions;
var getChildLogger = text_runtime_exports.getChildLogger;
var getDefaultRedactPatterns = text_runtime_exports.getDefaultRedactPatterns;
var getDiagnosticSessionStateCountForTest = text_runtime_exports.getDiagnosticSessionStateCountForTest;
var getLogger = text_runtime_exports.getLogger;
var getResolvedLoggerSettings = text_runtime_exports.getResolvedLoggerSettings;
var hasNonEmptyString = text_runtime_exports.hasNonEmptyString;
var hasOrphanReasoningCloseBoundary = text_runtime_exports.hasOrphanReasoningCloseBoundary;
var hasSystemMark = text_runtime_exports.hasSystemMark;
var isAutoLinkedFileRef = text_runtime_exports.isAutoLinkedFileRef;
var isFileLogLevelEnabled = text_runtime_exports.isFileLogLevelEnabled;
var isInsideCode = text_runtime_exports.isInsideCode;
var isRecord = text_runtime_exports.isRecord;
var localeLowercasePreservingWhitespace = text_runtime_exports.localeLowercasePreservingWhitespace;
var logActiveRuns = text_runtime_exports.logActiveRuns;
var logDebug = text_runtime_exports.logDebug;
var logError = text_runtime_exports.logError;
var logInfo = text_runtime_exports.logInfo;
var logLaneDequeue = text_runtime_exports.logLaneDequeue;
var logLaneEnqueue = text_runtime_exports.logLaneEnqueue;
var logMessageProcessed = text_runtime_exports.logMessageProcessed;
var logMessageQueued = text_runtime_exports.logMessageQueued;
var logRunAttempt = text_runtime_exports.logRunAttempt;
var logSessionStateChange = text_runtime_exports.logSessionStateChange;
var logSessionStuck = text_runtime_exports.logSessionStuck;
var logSuccess = text_runtime_exports.logSuccess;
var logToolLoopAction = text_runtime_exports.logToolLoopAction;
var logWarn = text_runtime_exports.logWarn;
var logWebhookError = text_runtime_exports.logWebhookError;
var logWebhookProcessed = text_runtime_exports.logWebhookProcessed;
var logWebhookReceived = text_runtime_exports.logWebhookReceived;
var lowercasePreservingWhitespace = text_runtime_exports.lowercasePreservingWhitespace;
var markdownToIR = text_runtime_exports.markdownToIR;
var markdownToIRWithMeta = text_runtime_exports.markdownToIRWithMeta;
var normalizeArrayBackedTrimmedStringList = text_runtime_exports.normalizeArrayBackedTrimmedStringList;
var normalizeAtHashSlug = text_runtime_exports.normalizeAtHashSlug;
var normalizeCsvOrLooseStringList = text_runtime_exports.normalizeCsvOrLooseStringList;
var normalizeE164 = text_runtime_exports.normalizeE164;
var normalizeFastMode = text_runtime_exports.normalizeFastMode;
var normalizeHyphenSlug = text_runtime_exports.normalizeHyphenSlug;
var normalizeLowercaseStringOrEmpty = text_runtime_exports.normalizeLowercaseStringOrEmpty;
var normalizeNullableString = text_runtime_exports.normalizeNullableString;
var normalizeOptionalLowercaseString = text_runtime_exports.normalizeOptionalLowercaseString;
var normalizeOptionalString = text_runtime_exports.normalizeOptionalString;
var normalizeOptionalStringifiedId = text_runtime_exports.normalizeOptionalStringifiedId;
var normalizeOptionalThreadValue = text_runtime_exports.normalizeOptionalThreadValue;
var normalizeOptionalTrimmedStringList = text_runtime_exports.normalizeOptionalTrimmedStringList;
var normalizeSingleOrTrimmedStringList = text_runtime_exports.normalizeSingleOrTrimmedStringList;
var normalizeStringEntries = text_runtime_exports.normalizeStringEntries;
var normalizeStringEntriesLower = text_runtime_exports.normalizeStringEntriesLower;
var normalizeStringifiedOptionalString = text_runtime_exports.normalizeStringifiedOptionalString;
var normalizeTrimmedStringList = text_runtime_exports.normalizeTrimmedStringList;
var parseInlineDirectives = text_runtime_exports.parseInlineDirectives;
var pathExists = text_runtime_exports.pathExists;
var prefixSystemMessage = text_runtime_exports.prefixSystemMessage;
var readStringField = text_runtime_exports.readStringField;
var readStringValue = text_runtime_exports.readStringValue;
var redactIdentifier = text_runtime_exports.redactIdentifier;
var redactSensitiveLines = text_runtime_exports.redactSensitiveLines;
var redactSensitiveText = text_runtime_exports.redactSensitiveText;
var redactToolDetail = text_runtime_exports.redactToolDetail;
var renderMarkdownIRChunksWithinLimit = text_runtime_exports.renderMarkdownIRChunksWithinLimit;
var renderMarkdownWithMarkers = text_runtime_exports.renderMarkdownWithMarkers;
var resetDiagnosticStateForTest = text_runtime_exports.resetDiagnosticStateForTest;
var resetLogger = text_runtime_exports.resetLogger;
var resolveConfigDir = text_runtime_exports.resolveConfigDir;
var resolveGlobalMap = text_runtime_exports.resolveGlobalMap;
var resolveGlobalSingleton = text_runtime_exports.resolveGlobalSingleton;
var resolveHomeDir = text_runtime_exports.resolveHomeDir;
var resolvePrimaryStringValue = text_runtime_exports.resolvePrimaryStringValue;
var resolveReactionLevel = text_runtime_exports.resolveReactionLevel;
var resolveRedactOptions = text_runtime_exports.resolveRedactOptions;
var resolveStuckSessionWarnMs = text_runtime_exports.resolveStuckSessionWarnMs;
var resolveUserPath = text_runtime_exports.resolveUserPath;
var safeParseJson = text_runtime_exports.safeParseJson;
var sanitizeAssistantVisibleText = text_runtime_exports.sanitizeAssistantVisibleText;
var sanitizeAssistantVisibleTextWithOptions = text_runtime_exports.sanitizeAssistantVisibleTextWithOptions;
var sanitizeAssistantVisibleTextWithProfile = text_runtime_exports.sanitizeAssistantVisibleTextWithProfile;
var sanitizeReplyDirectiveId = text_runtime_exports.sanitizeReplyDirectiveId;
var sanitizeTerminalText = text_runtime_exports.sanitizeTerminalText;
var setLoggerOverride = text_runtime_exports.setLoggerOverride;
var sha256HexPrefix = text_runtime_exports.sha256HexPrefix;
var shortenHomeInString = text_runtime_exports.shortenHomeInString;
var shortenHomePath = text_runtime_exports.shortenHomePath;
var sleep = text_runtime_exports.sleep;
var sliceMarkdownIR = text_runtime_exports.sliceMarkdownIR;
var sliceUtf16Safe = text_runtime_exports.sliceUtf16Safe;
var startDiagnosticHeartbeat = text_runtime_exports.startDiagnosticHeartbeat;
var stopDiagnosticHeartbeat = text_runtime_exports.stopDiagnosticHeartbeat;
var stripAssistantInternalScaffolding = text_runtime_exports.stripAssistantInternalScaffolding;
var stripDowngradedToolCallText = text_runtime_exports.stripDowngradedToolCallText;
var stripInlineDirectiveTagsForDelivery = text_runtime_exports.stripInlineDirectiveTagsForDelivery;
var stripInlineDirectiveTagsForDisplay = text_runtime_exports.stripInlineDirectiveTagsForDisplay;
var stripInlineDirectiveTagsFromMessageForDisplay = text_runtime_exports.stripInlineDirectiveTagsFromMessageForDisplay;
var stripMarkdown = text_runtime_exports.stripMarkdown;
var stripMinimaxToolCallXml = text_runtime_exports.stripMinimaxToolCallXml;
var stripReasoningTagsFromText = text_runtime_exports.stripReasoningTagsFromText;
var stripToolCallXmlTags = text_runtime_exports.stripToolCallXmlTags;
var summarizeStringEntries = text_runtime_exports.summarizeStringEntries;
var toPinoLikeLogger = text_runtime_exports.toPinoLikeLogger;
var truncateUtf16Safe = text_runtime_exports.truncateUtf16Safe;
var withTimeout = text_runtime_exports.withTimeout;
export { CONFIG_DIR, DEFAULT_LOG_DIR, DEFAULT_LOG_FILE, DEFAULT_WEB_MEDIA_BYTES, DEFAULT_WHATSAPP_MEDIA_MAX_MB, FILE_REF_EXTENSIONS_WITH_TLD, SYSTEM_MARK, WHATSAPP_LEGACY_OUTBOUND_SEND_DEP_KEYS, __test__, asNullableObjectRecord, asNullableRecord, asOptionalObjectRecord, asOptionalRecord, asRecord, assertWebChannel, bindAbortRelay, buildTimeoutAbortSignal, chunkItems, chunkMarkdownIR, clamp, clampInt, clampNumber, convertMarkdownTables, createScopedExpiringIdCache, diagnosticLogger, displayPath, displayString, ensureDir, escapeRegExp, fetchWithTimeout, findCodeRegions, getChildLogger, getDefaultRedactPatterns, getDiagnosticSessionStateCountForTest, getLogger, getResolvedLoggerSettings, hasAnyWhatsAppAuth, hasNonEmptyString, hasOrphanReasoningCloseBoundary, hasSystemMark, isAutoLinkedFileRef, isFileLogLevelEnabled, isInsideCode, isRecord, isSelfChatMode, isWhatsAppGroupJid, isWhatsAppUserTarget, jidToE164, listEnabledWhatsAppAccounts, listAccountIds as listWhatsAppAccountIds, listWhatsAppAuthDirs, listWhatsAppDirectoryGroupsFromConfig, listWhatsAppDirectoryPeersFromConfig, localeLowercasePreservingWhitespace, logActiveRuns, logDebug, logError, logInfo, logLaneDequeue, logLaneEnqueue, logMessageProcessed, logMessageQueued, logRunAttempt, logSessionStateChange, logSessionStuck, logSuccess, logToolLoopAction, logWarn, logWebhookError, logWebhookProcessed, logWebhookReceived, looksLikeWhatsAppTargetId, lowercasePreservingWhitespace, markdownToIR, markdownToIRWithMeta, markdownToWhatsApp, normalizeArrayBackedTrimmedStringList, normalizeAtHashSlug, normalizeCsvOrLooseStringList, normalizeE164, normalizeFastMode, normalizeHyphenSlug, normalizeLowercaseStringOrEmpty, normalizeNullableString, normalizeOptionalLowercaseString, normalizeOptionalString, normalizeOptionalStringifiedId, normalizeOptionalThreadValue, normalizeOptionalTrimmedStringList, normalizeSingleOrTrimmedStringList, normalizeStringEntries, normalizeStringEntriesLower, normalizeStringifiedOptionalString, normalizeTrimmedStringList, normalizeWhatsAppAllowFromEntries, normalizeWhatsAppMessagingTarget, normalizeWhatsAppTarget, parseInlineDirectives, pathExists, prefixSystemMessage, readStringField, readStringValue, redactIdentifier, redactSensitiveLines, redactSensitiveText, redactToolDetail, renderMarkdownIRChunksWithinLimit, renderMarkdownWithMarkers, resetDiagnosticStateForTest, resetLogger, resolveConfigDir, resolveDefaultWhatsAppAccountId, resolveGlobalMap, resolveGlobalSingleton, resolveHomeDir, resolveJidToE164, resolvePrimaryStringValue, resolveReactionLevel, resolveRedactOptions, resolveStuckSessionWarnMs, resolveUserPath, resolveWhatsAppAccount, resolveWhatsAppAuthDir, resolveWhatsAppGroupIntroHint, resolveWhatsAppGroupRequireMention, resolveWhatsAppGroupToolPolicy, resolveWhatsAppMediaMaxBytes, resolveWhatsAppOutboundTarget, safeParseJson, sanitizeAssistantVisibleText, sanitizeAssistantVisibleTextWithOptions, sanitizeAssistantVisibleTextWithProfile, sanitizeReplyDirectiveId, sanitizeTerminalText, setLoggerOverride, sha256HexPrefix, shortenHomeInString, shortenHomePath, sleep, sliceMarkdownIR, sliceUtf16Safe, startDiagnosticHeartbeat, stopDiagnosticHeartbeat, stripAssistantInternalScaffolding, stripDowngradedToolCallText, stripInlineDirectiveTagsForDelivery, stripInlineDirectiveTagsForDisplay, stripInlineDirectiveTagsFromMessageForDisplay, stripMarkdown, stripMinimaxToolCallXml, stripReasoningTagsFromText, stripToolCallXmlTags, summarizeStringEntries, toPinoLikeLogger, toWhatsappJid, truncateUtf16Safe, __testing as whatsappAccessControlTesting, whatsappCommandPolicy, whatsappPlugin, whatsappSetupPlugin, withTimeout };
